### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by tiyash-basu-sociomantic**

### CODE REVIEW

File: src/swarm/neo/authentication/HmacAuthCode.d [hunk 1]
```diff
@@ -109,10 +109,7 @@ struct HmacAuthCode
     import cstdio = core.stdc.stdio: fprintf, stderr;
     import core.stdc.stdlib: exit, EXIT_FAILURE;
 
-    version (D_Version2)
-        mixin("shared static this ( ) { staticCtor(); }");
-    else
-        static this ( ) { staticCtor(); }
+    shared static this ( ) { staticCtor(); }
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/authentication/HmacAuthCode.d]: This could be refactored to include the code that's inside `staticCtor`. I don't think having that separate function is needed any more.

File: src/swarm/neo/node/Connection.d [hunk 2]
```diff
@@ -23,7 +23,7 @@ class Connection: ConnectionBase
     import swarm.neo.node.RequestSet;
     import swarm.neo.protocol.connect.NodeConnect;
     import swarm.neo.authentication.HmacDef: Key;
-    import swarm.neo.connection.YieldedRequestOnConns;
+    public import swarm.neo.connection.YieldedRequestOnConns;
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/node/Connection.d]: Did you copy this decision (to use `public import`) from somewhere else (e.g. ocean)? It"s something we've traditionally avoided, but I may have missed that it's now being used in certain circumstances.

File: src/swarm/neo/client/ConnectionSet.d [hunk 3]
```diff
@@ -917,12 +918,8 @@ version (UnitTest)
 unittest
 {
     static void setAddress ( ref AddrPort address, ubyte[] address_bytes ... )
-    in
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/ConnectionSet.d:920]: Are these changes related to the D2 transition?

File: src/swarm/client/registry/DisablableNodeSet.d [hunk 4]
```diff
@@ -83,16 +83,14 @@ public class DisablableNodeSet : NodeSet
     ***************************************************************************/
 
     public void disable ( NodeItem node )
-    out
```
Comment (gavin-norman-sociomantic) [inline src/swarm/client/registry/DisablableNodeSet.d]: Are these changes related to the D2 transition?

File: src/swarm/neo/node/IRequest.d [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    reason for this is documented in `ConnectionHandler.emplace`.
+
+    copyright: Copyright (c) 2018 dunnhumby Germany GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module swarm.neo.node.IRequest;
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/node/IRequest.d:16]: This rename is a breaking change and should have a release note (for the sanity of library users).

**Review by gavin-norman-sociomantic — COMMENTED** [review]
A few comments, but looks good!

File: src/swarm/neo/node/Connection.d (hunk 2 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/node/Connection.d]: This was not copied from any other project. I did not know that we are avoiding this.

Would you prefer to not use `public imports` here?

File: src/swarm/client/registry/DisablableNodeSet.d (hunk 4 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/client/registry/DisablableNodeSet.d]: This was mentioned in the D2-migration docs [here](https://github.com/sociomantic/backend/blob/master/doc/d2-migration/stage3.md#invariants).

File: src/swarm/neo/client/ConnectionSet.d (hunk 3 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/client/ConnectionSet.d]: This was mentioned in the D2-migration docs [here](https://github.com/sociomantic/backend/blob/master/doc/d2-migration/stage3.md#invariants).

File: src/swarm/neo/authentication/HmacAuthCode.d (hunk 1 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/authentication/HmacAuthCode.d]: I added a release note for this.

File: src/swarm/neo/node/IRequest.d (hunk 5 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/node/IRequest.d:16]: I added a release note for this.

File: src/swarm/neo/authentication/HmacAuthCode.d (hunk 1 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/authentication/HmacAuthCode.d]: Done!

File: src/swarm/client/registry/DisablableNodeSet.d (hunk 4 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/client/registry/DisablableNodeSet.d]: The more detailed doc about assert/enforce/verify says something different: https://github.com/sociomantic-tsunami/sociomantic/blob/master/Code/D/assert-vs-enforce.md#adapting-older-code
>All checks in in contracts should be converted to verify statements in the method/function body.
>out contracts and invariants can remain as they are, using asserts. They will not be present in release builds.

The weird thing is that we already did a sweep of removing all `in` contracts from swarm, so I'm confused as to how you've discovered some!

File: src/swarm/neo/node/Connection.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/node/Connection.d]: As far as I remember, in other library locations where we came across this, we added the imports in the locations where they're actually used.

File: src/swarm/client/registry/DisablableNodeSet.d (hunk 4 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/client/registry/DisablableNodeSet.d]: Interesting.. I have found some commits that remove `in` contracts, but the ones changed in this PR were not touched back then. Maybe they were missed for some reason?

Also, the `out` contracts have been put back now.

**Comment by tiyash-basu-sociomantic** [comment]
Marked #390 to be fixed in the commit titled `Remove D1 tests from CI`.

File: src/swarm/neo/protocol/socket/MessageReceiver.d [hunk 6]
```diff
@@ -383,17 +383,15 @@ private class MessageReceiverBase
...
-        assert(this); // invariant
-    }
     out
     {
         assert(bytes_requested <= this.buffer_content_end);
         assert(this); // invariant
     }
     body
     {
+        verify(this !is null);
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/protocol/socket/MessageReceiver.d]: This is wrong. Doing `assert(this);` causes the invariant to be called. (Hence the comment.)

File: src/swarm/neo/protocol/socket/MessageReceiver.d [hunk 7]
```diff
@@ -515,16 +513,14 @@ class MessageReceiver: MessageReceiverBase
...
-    {
-        assert(this);
-    }
     out
     {
         assert(this);
     }
     body
     {
+        verify(this !is null);
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/protocol/socket/MessageReceiver.d]: Ditto.

File: src/swarm/client/registry/DisablableNodeSet.d (hunk 4 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/client/registry/DisablableNodeSet.d]: It _could_ be that "in contracts (containing asserts) should only be used on private methods/ functions which are called via a public API including verify/enforce checks. in contracts on non-private methods/functions should be avoided." applies to them? (https://github.com/sociomantic-tsunami/sociomantic/blob/master/Code/D/assert-vs-enforce.md#general-error-checking-principles)

File: src/swarm/neo/node/Connection.d (hunk 2 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/node/Connection.d]: I have reverted back to `import` from `public import` in most cases, and imported modules explicitly, wherever needed.

In some places in `swarm`, I noticed that `public import`s are already being used, e.g., in `swarm.util.Verify`. I also found a few instances in `ocean`. So, in the [commit in this pr](https://github.com/sociomantic-tsunami/swarm/pull/393/commits/23d6a2b93abe9ac1907f4d17b9a53bbdafb7ee05), `public import`s are still introduced in some places in tests, where adding them keeps the design simplified.

File: src/swarm/neo/protocol/socket/MessageReceiver.d (hunk 6 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/protocol/socket/MessageReceiver.d]: Thanks for pointing out. Replaced the invariant call with the invariant body here.

File: src/swarm/neo/protocol/socket/MessageReceiver.d (hunk 7 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/protocol/socket/MessageReceiver.d]: Replaced the invariant call with the invariant body here.

**Comment by matthias-wende-sociomantic** [comment]
Does all of this have to happen in one PR. Otherwise I'd suggest to have one PR for pure D2 migration and a succeeding cleanup PR.

**Comment by gavin-norman-sociomantic** [comment]
That's what I was wondering. We could have a PR for the D2 stuff, then a PR for the deprecations, say.

File: src/swarm/neo/client/RequestSet.d [hunk 8]
```diff
@@ -606,7 +606,13 @@ public final class RequestSet: IRequestSet
         private void reset ( )
         out
         {
-            assert(this); // invariant
+            if (!this.id)
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/RequestSet.d]: Wait, what's the reason for this?

File: src/swarm/client/request/params/IRequestParams.d [hunk 9]
```diff
@@ -538,7 +538,6 @@ public abstract class IRequestParams
         ***********************************************************************/
 
         static typeof(&this) serialized ( void[] data )
-        in
```
Comment (gavin-norman-sociomantic) [inline src/swarm/client/request/params/IRequestParams.d:541]: The commit message for this commit (" Move `in` contracts within function bodies ") is a bit inaccurate. Maybe this?
"
Move `in` contracts within function bodies
All `in` contracts were converted into `verify` calls in method bodies in a previous PR. It seems some were missed. Converting these now. (The reason for converting `in` contracts to `verify` calls is to make important checks also occur in release builds.)
"

File: src/swarm/neo/client/RequestSet.d (hunk 8 above)
Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/client/RequestSet.d]: Thanks for pointing out. It seems like I missed it. I'll do another sweep.

Comment (tiyash-basu-sociomantic) [reply src/swarm/neo/client/RequestSet.d]: Ah, I see. I copied the contents of the invariant into the out block, so that checks specified here are not removed.

**Comment by tiyash-basu-sociomantic** [comment]
As discussed, this PR has been split into two. This PR takes care of some optimizations post-D2 migration, and removing deprecations.

ATM, it's blocked on https://github.com/sociomantic-tsunami/swarm/pull/395 to drop D1 support.

**Comment by tiyash-basu-sociomantic** [comment]
Rebased on the `v6.x.x` branch.

File: src/swarm/neo/protocol/socket/MessageReceiver.d (hunk 7 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/protocol/socket/MessageReceiver.d]: Would be nice to not make this intermediary wrong change here.

File: src/swarm/neo/protocol/socket/MessageReceiver.d (hunk 6 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/protocol/socket/MessageReceiver.d]: Would be nice to not make this intermediary wrong change here.

File: src/swarm/neo/client/RequestSet.d (hunk 8 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/RequestSet.d]: Why though? (This isn't what the assert/enforce/verify doc recommends doing. Unless I'm going mad.)

**Review by gavin-norman-sociomantic — COMMENTED** [review]
All commits which I've not commented on look good.

**Comment by tiyash-basu-sociomantic** [comment]
As discussed, this commit refactoring `out` contracts has been removed, so that invariants are still executed in non-release builds.

**Review by gavin-norman-sociomantic — APPROVED** [review]
